laceworksdk/http_session.py

```diff
@@ -220,115 +248,141 @@ def get(self, uri, org=False):
 
         return response
 
-    def patch(self, uri, org=False, data=None, param=None):
+    def get(self, uri, params=None, **kwargs):
         """
-        :param uri: uri to send the HTTP POST request to
-        :param org: boolean representing whether the request should be performed at the Organization level
-        :param data: json object containing the data
-        :param param: python object containing the parameters
+        :param uri: uri to send the HTTP GET request to
+        :param params: dict of parameters for the HTTP request
+        :param kwargs: passed on to the requests package
 
         :return: response json
 
-        :raises: ApiError if unable to get a connection
+        :raises: ApiError if anything but expected response code is returned
         """
 
-        self._check_access_token()
+        # Perform a GET request
+        response = self._request("GET", uri, params=params, **kwargs)
 
-        uri = f"{self._base_url}{uri}"
+        return response
 
-        logger.info(f"PATCH request to URI: {uri}")
-        logger.debug(f"PATCH request data:\n{data}")
+    def get_pages(self, uri, params=None, **kwargs):
+        """
+        :param uri: uri to send the initial HTTP GET request to
+        :param params: dict of parameters for the HTTP request
+        :param kwargs: passed on to the requests package
 
-        # Perform a PATCH request
-        response = self._session.patch(uri, params=param, json=data, headers=self._get_request_headers(org_access=org))
+        :return: a generator that Gets and yields pages of data
 
-        # Validate the response
-        self._check_response_code(response, DEFAULT_SUCCESS_RESPONSE_CODES)
+        :raises: ApiError if anything but expected response code is returned
+        """
 
-        self._print_debug_response(response)
+        response = self.get(uri, params=params, **kwargs)
 
-        return response
+        while True:
+            yield response
+
+            try:
+                next_page = response.json().get('paging').get('urls').get('nextPage')
+            except Exception:
+                next_page = None
+
+            if next_page:
+                response = self.get(uri, params=params, **kwargs)
```

Comment:
How do we communicate `next_page` to this request?

Reply:
Great catch!  resolved in latest commits - still waiting on some official docs to be released, and more endpoints that actually hit the 5000 row limit so that I can finish testing